### DISCUSSION TITLE
What is the recommended way of doing "--data-raw" with reastease?

### DISCUSSION THREAD

**Discussion by mifriis** [canton7/RestEase#204] [Q&A] [answered]
Say I am interested in doing a request similar to:
`curl -X POST -H "Content-Type: text/plain" --data-raw "3" http://somewebsite.sum/someapi/lenny`

As in posting some raw data in the body. Something that isn't JSON.

Would the equivalent in RestEase be something like:

```C#
[Post("/someApi/{name}")]
Task SetNumber([Path] string name, [Body] int number);
```

**Comment by canton7** [comment, marked as answer]
[See here](https://github.com/canton7/RestEase#body-content).

If you pass a string, that value is used directly as the body. If you pass a different value (which isn't one of the other special ones in that list), it will be serialized as JSON using Json.NET.

Json.NET serializes the object `3` as the string `3`, so what you've posted should do what you expect. However in general if you want to control exactly what goes in the body, make it a `string`.
